platforms/jvm/language-java/src/integTest/groovy/org/gradle/api/tasks/compile/JavaCompileProblemsIntegrationTest.groovy

```diff
@@ -153,13 +153,33 @@ class JavaCompileProblemsIntegrationTest extends AbstractIntegrationSpec impleme
             assertLocations(it, true)
             severity == Severity.WARNING
             fqid == 'compilation:java:compiler-warn-redundant-cast'
-            contextualLabel == 'redundant cast to java.lang.String'
+            if (JavaVersion.current().java9Compatible) {
+                // Above Java 9, we can use the RichDiagnosticFormatter
+                contextualLabel == 'redundant cast to String'
+            } else {
+                // Below Java 9, we have to use the BasicDiagnosticFormatter
+                contextualLabel == 'redundant cast to java.lang.String'
+            }
+            contextualLabel.split(System.lineSeparator()).length == 1
```

Comment:
I think a lot of these are redundant/unnecessary because we're checking that the contextualLabel is exactly a particular string and that string does not contain new lines.  That means this check will never fail on its own.  I would remove this in all places where we're checking for an exact string.

Reply:
That is a really good point, I agree